src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -448,20 +475,29 @@
         this)
       (a/stop [this]
         (shutdown! app-context)
-        this))))
+        this)
+      (a/restart [this]
+        (try
+          (run-lifecycle-fns app-context s/stop "stop" (reverse ordered-services))
```

Comment:
@cprice404 Fair enough.  May just need to document this as an expectation for services, then.  Arguably a "breaking change", I suppose, although it would only come into play when a new feature, HUP support, gets used.  So semantically may be okay to keep Trapperkeeper on major version 1 after this PR comes in.

So if we're willing to live with the double-stop behavior, not sure it's worth the effort to have the implementation here call `stop` on other services after the first one fails since the tk-main shutdown processing would ensure later that every service's `stop` is called at least one time.